### PR TITLE
Add rule for genome indexing with STAR

### PR DESCRIPTION
**Pull request by simonlevine**
Added STAR index stage from STAR website's reference genome. Unsure if we need the .gtf file.

### CODE REVIEW

**Comment by simonlevine** [comment]
"{gtf} "  # Highly recommended GTF
 per the docs

**Comment by jeremyadamsfisher** [comment]
General thoughts:
- Instead of downloading in one rule and using in another, you can combine those in one rule
- No need for wildcards here, replace `{thing}` with the actual file name (we need not generalize to allow arbitrary genomes -- that feels overly complicated, no?)
- Output of this rule should flow to the input of the `star_single_ended` rule

File: Snakefile [hunk 1]
```diff
@@ -22,6 +21,23 @@ rule download_genome:
...
+        fasta = "data/raw/reference_genome/{genome}.fa"
+        gtf = 'data/raw/reference_genome/{gtf}.gtf'
+    output:
+        directory("{genome}")
+    message:
+        "Running STAR index"
+    threads:
+        1
+    params:
+        extra = ""
```
Comment (jeremyadamsfisher) [inline Snakefile:22]: This is redundant

File: Snakefile [hunk 2]
```diff
@@ -22,6 +21,23 @@ rule download_genome:
...
+rule star_index:
+    input:
+        fasta = "data/raw/reference_genome/{genome}.fa"
+        gtf = 'data/raw/reference_genome/{gtf}.gtf'
+    output:
+        directory("{genome}")
+    message:
+        "Running STAR index"
+    threads:
+        1
```
Comment (jeremyadamsfisher) [inline Snakefile]: We have like 128 cores, no need to enforce single threading here

**Pull request merged**